state/execution.go

```diff
@@ -37,9 +38,11 @@ type BlockExecutor struct {
 	mempool mempl.Mempool
 	evpool  EvidencePool
 
-	logger log.Logger
-
+	logger  log.Logger
 	metrics *Metrics
+
+	// cache the verification results over a single height
+	cache []bytes.HexBytes
```

Comment:
why not consider `map[string]struct{}`? Granted, there is overhead for `hash.String()`, but it's cleaner and possibly more performant and an iterative search.

Reply:
Yeah I had considered it was pretty indifferent between the two. I'll make the change